src/components/modal.vue

```diff
@@ -0,0 +1,97 @@
+<template>
+  <transition name="modal">
+    <div v-if="is_open" id="modal-mask" @click.self="close">
```

Comment:
Can the "close when clicking outside of the modal" behavior be optional? There are times where we want the user to have to click on one of the options in the modal (like when confirming a submit). I think the default should be that it does NOT close when user clicks outside.